File: src/editors/blur.ts
import * as sharp from "sharp"

export default {
    name: "Blur",
    description: "Allows you to blur a image.",
    icon: "blur.png",
    expectsImageData: true,
    apply: (partBuff: Buffer): Promise<Buffer> => sharp(partBuff).blur(10).toBuffer(),
}

File: core/editors/blur.go
package editors

import (
	"github.com/disintegration/gift"
	"github.com/magiccap/MagicCap/assets/editor"
	"image"
)

func init() {
	Editors["blur"] = &Editor{
		Name:        "Blur",
		Description: "Allows you to blur a image.",
		Icon:        editor.Blur,
		Apply: func(Region *image.RGBA, _ [3]uint8) *image.RGBA {
			g := gift.New(
				gift.GaussianBlur(20),
			)
			img := image.NewRGBA(Region.Bounds())
			g.Draw(img, Region)
			return img
		},
	}
}
